### PR TITLE
feat(Icon): update TS  structure

### PR DESCRIPTION
**Pull request by soumyaloka**
## Description

This PR updates Icon TS file to new TS template structure

## References

LUI-970

### CODE REVIEW

**Comment by anthony9187** [comment]
since `color` is in the properties array it should also be included in the TemplateSpec and class declaration, because it can still be set via the template or `Icon.color = newcolor`

**Review by arwehrman — APPROVED** [review]
only comment is the same Anthony had for color

**Review by arwehrman — APPROVED** [review]

File: packages/@lightningjs/ui-components/src/components/Icon/Icon.d.ts [hunk 1]
```diff
@@ -16,16 +16,63 @@
...
 };
 
-export default class Icon extends Base {
-  icon: string;
+declare namespace Icon {
+  export interface TemplateSpec extends Base.TemplateSpec {
+    /**
+     * icon color (can only be applied on png icons)
+     */
+    color: number; //TODO: return type of color is string or number and is it optional
```
Comment (anthony9187) [inline packages/@lightningjs/ui-components/src/components/Icon/Icon.d.ts]: this should be type Color not number

File: packages/@lightningjs/ui-components/src/components/Icon/Icon.d.ts [hunk 2]
```diff
@@ -16,16 +16,63 @@
...
+}
+
+declare class Icon<
+  TemplateSpec extends Icon.TemplateSpec = Icon.TemplateSpec,
+  TypeConfig extends lng.Component.TypeConfig = lng.Component.TypeConfig
+> extends Base<TemplateSpec, TypeConfig> {
+  /**
+   * icon color (can only be applied on png icons)
+   */
+  color: number;
```
Comment (anthony9187) [inline packages/@lightningjs/ui-components/src/components/Icon/Icon.d.ts]: same here

**Review by anthony9187 — APPROVED** [review]

**Pull request merged**